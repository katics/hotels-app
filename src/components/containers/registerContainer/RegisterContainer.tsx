import { ChangeEvent, FC, useEffect, useState } from "react";
import { useHistory } from "react-router";

import store from "../../../store";
import HotelProvider, {
  hotelContext,
  useDispatchHotel,
  useSelectorHotel,
} from "../../../store/HotelProvider";
import { fetchRegisterUser } from "../../../store/user/userActions";
import { userSeletor } from "../../../store/user/userSelector";
import { APP_ROUTES } from "../../../utils/AppRoutes";
import { UserRegistrationData } from "../../../utils/types/UserRegistrationData";
import UserRegistrationPage from "../../pages/registerPage/RegisterPage";
import Spinner from "../../spinner/Spinner";

const UserRegistrationContainer: FC = () => (
  <HotelProvider store={store} context={hotelContext}>
    <UserRegistration />
  </HotelProvider>
);

const UserRegistration: FC = () => {
  const history = useHistory();
  const userDispatch = useDispatchHotel();
  const { registerError, hasRegisred, isLoading } = useSelectorHotel(
    userSeletor
  );

  const userRegistrationData: UserRegistrationData = {
    username: "",
    first_name: "",
    password: "",
    confirmPassword: "",
    last_name: "",
    email: "",
  };

  const [registrationData, setRegistrationData] = useState(
    userRegistrationData
  );

  const signInClick = (e: ChangeEvent<HTMLFormElement>) => {
    e.preventDefault();
    console.log(registrationData);
    userDispatch(fetchRegisterUser(registrationData));
  };

  const userNameChanged = (e: any) => {
    setRegistrationData({
      ...registrationData,
      username: e.target.value,
    });
  };

  const firstNameChanged = (e: any) => {
    setRegistrationData({
      ...registrationData,
      first_name: e.target.value,
    });
  };

  const lastNameChanged = (e: any) => {
    setRegistrationData({
      ...registrationData,
      last_name: e.target.value,
    });
  };

  const emailChanged = (e: any) => {
    setRegistrationData({
      ...registrationData,
      email: e.target.value,
    });
  };

  const passwordChanged = (e: any) => {
    setRegistrationData({
      ...registrationData,
      password: e.target.value,
    });
  };

  const confirmPasswordChanged = (e: any) => {
    setRegistrationData({
      ...registrationData,
      confirmPassword: e.target.value,
    });
  };
  useEffect(() => {
    if (hasRegisred) history.push(APP_ROUTES.loginPage);
  }, [hasRegisred]);
  return (
    <>
      {isLoading ? (
        <Spinner />
      ) : (
        <UserRegistrationPage
          registerError={registerError}
          registrationData={registrationData}
          signInClick={signInClick}
          userNameChanged={userNameChanged}
          firstNameChanged={firstNameChanged}
          lastNameChanged={lastNameChanged}
          emailChanged={emailChanged}
          passwordChanged={passwordChanged}
          confirmPasswordChanged={confirmPasswordChanged}
        />
      )}
    </>
  );
};

export default UserRegistrationContainer;
