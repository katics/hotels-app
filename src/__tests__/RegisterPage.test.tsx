import RegisterContainer from "../components/containers/registerContainer/RegisterContainer";
import { StaticRouter } from "react-router-dom";

import { render } from "@testing-library/react";

describe("Register Page", () => {
  it("Should match register page snapshot", () => {
    const { container } = render(
      <StaticRouter>
        <RegisterContainer />
      </StaticRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
