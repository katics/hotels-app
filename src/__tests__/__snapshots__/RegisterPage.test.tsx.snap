// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Page Should match register page snapshot 1`] = `
<div
  class="container-fluid"
>
  <div
    class="row justify-content-center"
  >
    <div
      class="col-lg-6"
    >
      <div
        class="card"
      >
        <h2
          class="card-title text-center"
        >
          Register
        </h2>
        <div
          class="card-body py-md-4"
        >
          <form
            autocomplete="off"
          >
            <div
              class="form-group"
            >
              <label>
                User Name
              </label>
              <input
                class="form-control"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label>
                First Name
              </label>
              <input
                class="form-control"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label>
                Last Name
              </label>
              <input
                class="form-control"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label>
                Email
              </label>
              <input
                class="form-control"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label>
                Password
              </label>
              <input
                class="form-control"
                type="password"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label>
                Confirm Password
              </label>
              <input
                class="form-control"
                type="password"
                value=""
              />
            </div>
            <div
              class="d-flex flex-row align-items-center justify-content-between"
            >
              <button
                class="btn btn-primary btn-block"
              >
                Create Account
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
